src/overlays/actors/ovl_En_An/z_en_an.c

```diff
@@ -13,160 +20,3438 @@
+    }
+
+    if (this->unk_3C0) {
+        scheduleOutput.result = ANJU_SCH_WITH_KAFEI;
+        if (this->scheduleResult != ANJU_SCH_WITH_KAFEI) {
+            EnAn_ProcessScheduleOutput(this, play, &scheduleOutput);
+        }
+    } else if (!Schedule_RunScript(play, sScheduleScript, &scheduleOutput) ||
+               ((this->scheduleResult != scheduleOutput.result) &&
+                !EnAn_ProcessScheduleOutput(this, play, &scheduleOutput))) {
+        this->actor.shape.shadowDraw = NULL;
+        this->actor.flags &= ~ACTOR_FLAG_TARGETABLE;
+        scheduleOutput.result = ANJU_SCH_NONE;
+    } else {
+        this->actor.shape.shadowDraw = ActorShadow_DrawCircle;
+        this->actor.flags |= ACTOR_FLAG_TARGETABLE;
+    }
+
+    this->scheduleResult = scheduleOutput.result;
+    this->lookAtActor = EnAn_FindLookAtActor(this, play);
+    EnAn_HandleSchedule(this, play);
+}
+
+void EnAn_Talk(EnAn* this, PlayState* play) {
+    if (func_8010BF58(&this->actor, play, this->msgEventScript, this->msgEventFunc, &this->msgScriptResumePos)) {
+        // Message event script is done
+
+        SubS_SetOfferMode(&this->stateFlags, SUBS_OFFER_MODE_ONSCREEN, SUBS_OFFER_MODE_MASK);
+
+        this->stateFlags &= ~ENAN_STATE_ENGAGED;
+        this->stateFlags |= ENAN_STATE_LOST_ATTENTION;
+        this->loseAttentionTimer = 20;
+        this->msgScriptResumePos = 0;
+        this->actionFunc = EnAn_FollowSchedule;
+
+        return;
+    }
+
+    if ((this->scheduleResult == ANJU_SCH_WAITING_FOR_KAFEI) || (this->scheduleResult == ANJU_SCH_LAUNDRY_POOL_SIT) ||
+        (this->scheduleResult == ANJU_SCH_RANCH) || (this->scheduleResult == ANJU_SCH_COOKING) ||
+        (this->scheduleResult == ANJU_SCH_WITH_KAFEI)) {
+        return;
+    }
+
+    if ((this->lookAtActor != NULL) && (this->lookAtActor->update != NULL)) {
+        // Rotate towards actor
+        s16 yaw;
+        Vec3f lookAtPos;
+        Vec3f pos;
+
+        Math_Vec3f_Copy(&lookAtPos, &this->lookAtActor->world.pos);
+        Math_Vec3f_Copy(&pos, &this->actor.world.pos);
+        yaw = Math_Vec3f_Yaw(&pos, &lookAtPos);
+
+        Math_ApproachS(&this->actor.shape.rot.y, yaw, 4, 0x2AA8);
+    }
+}
+
+void EnAn_HandleCouplesMaskCutscene(EnAn* this, PlayState* play) {
+    s32 csAnimIndex[] = {
+        /* 0 */ 0, // no cue
+        /* 1 */ ENAN_ANIM_SITTING_IN_DISBELIEF,
+        /* 2 */ ENAN_ANIM_LOOK_UP,
+        /* 3 */ ENAN_ANIM_MASK_STAND_LOOP,
+        /* 4 */ ENAN_ANIM_MASK_KNEEL,
+        /* 5 */ ENAN_ANIM_HUG,
+        /* 6 */ ENAN_ANIM_HUG_RELEASE,
+        /* 7 */ ENAN_ANIM_COMBINE_MASKS_1,
+        /* 8 */ ENAN_ANIM_COMBINE_MASKS_2,
+        /* 9 */ ENAN_ANIM_HOLD_HANDS,
+    };
+    s32 pad;
+
+    //! FAKE
+    if (0) {}
+
+    if (Cutscene_IsCueInChannel(play, CS_CMD_ACTOR_CUE_557)) {
+        s32 cueChannel = Cutscene_GetCueChannel(play, CS_CMD_ACTOR_CUE_557);
+        u16 cueId = play->csCtx.actorCues[cueChannel]->id;
+
+        if (this->cueId != (cueId & 0xFF)) {
```

Comment:
```suggestion
        if (this->cueId != (u8)cueId) {
```
Tends to be more common